src/measure_numbers_adjust_for_leadin.lua

```diff
@@ -41,64 +41,76 @@ function measure_numbers_adjust_for_leadin()
+                                        local sep_nums = finale.FCSeparateMeasureNumbers()
+                                        sep_nums:LoadAllInCell(cell)
+                                        if (sep_nums.Count > 0) then
+                                            for sep_num in each(sep_nums) do
+                                                sep_num.HorizontalPosition = -lead_in
+                                                sep_num:Save()
+                                            end
+                                        elseif (0 ~= lead_in) then
+                                            local sep_num = finale.FCSeparateMeasureNumber()
+                                            sep_num:ConnectCell(cell)
+                                            sep_num:AssignMeasureNumberRegion(meas_num_region)
                                             sep_num.HorizontalPosition = -lead_in
-                                            sep_num:Save()
-                                        end
-                                    elseif (0 ~= lead_in) then
-                                        local sep_num = finale.FCSeparateMeasureNumber()
-                                        sep_num:ConnectCell(cell)
-                                        sep_num:AssignMeasureNumberRegion(meas_num_region)
-                                        sep_num.HorizontalPosition = -lead_in
-                                        --sep_num:SetShowOverride(true) -- enable this line if you want to force show the number. otherwise it will show or hide based on the measure number region
-                                        if sep_num:SaveNew() then
-                                            local measure = finale.FCMeasure()
-                                            measure:Load(cell.Measure)
-                                            measure:SetContainsManualMeasureNumbers(true)
-                                            measure:Save()
+                                            --sep_num:SetShowOverride(true) -- enable this line if you want to force show the number. otherwise it will show or hide based on the measure number region
+                                            if sep_num:SaveNew() then
+                                                local measure = finale.FCMeasure()
+                                                measure:Load(cell.Measure)
+                                                measure:SetContainsManualMeasureNumbers(true)
+                                                measure:Save()
+                                            end
                                         end
                                     end
                                 end
                             end
                         end
                     end
+                    previous_meas_num = meas_num
                 end
             end
         end
```

Comment:
You aren't wrong, but I'm a bit reluctant to refactor it at the moment, because it's tested and works. Since you went ahead and merged, I'm guessing you must have reached the same conclusion. I really wish Lua had a `continue` statement. I suppose it has `goto` but I don't like those very much.

Reply:
Yeah, a `continue` keyword would be really nice…